### PR TITLE
Rip out drake/matlab/util from the drake/util directory

### LINKED ISSUE #3278: Carve out the non-matlab portions of `drake/util` into new homes
Related to #3253, we should find new homes for the C++ forward-looking portions of `drake/util`.  This has been happening incrementally (see prior `common` and `math` related PRs, such as #2666, #2779, #2785 #3041 #3121, and cleanups like #3166 #3179 #3180 #3276 #3277).  This issue can serve as the burn-down list for finishing up.

### PR DESCRIPTION
**Pull request by jwnimmer-tri**
Fixes #3278.  A ~~second~~ _third_ portion of issue #3253.  A second attempt at PR #3832.

### CODE REVIEW

**Comment by jwnimmer-tri** [comment]
Argh matlab flake!
@drake-jenkins-bot retest this please.

**Comment by jwnimmer-tri** [comment]
+@david-german-tri for both reviews, please.

This PR should be reviewed commit-at-a-time.

Commits 1 and 3 are the shell script that does the "git mv" operations.  Commit 2 is the result of running that script.  Commits 4+ are path fixups as a result of moved files.

I suggest that for this PR, we review using the direct github tooling; reviewable.io chokes a bit on the size.  Please review using the "Commits" tab above -- each commit is well-focused, skilling them on a per-type-of-edit basis is likely the most efficient.

If you decide to use reviewable instead, (1) use chrome, and (2) using the "File Matrix" click-and-drag-range to select one commit at a time to show diffs for.

During review, I will not rebase this PR.

**Comment by david-german-tri** [comment]
Starting review now, as suggested above.

**Comment by david-german-tri** [comment]
@drake-jenkins-bot linux-gcc-ninja-experimental please
@drake-jenkins-bot linux-gcc-ninja-experimental-matlab please
@drake-jenkins-bot mac-clang-experimental please
@drake-jenkins-bot mac-clang-experimental-matlab please
@drake-jenkins-bot mac-clang-ninja-experimental please
@drake-jenkins-bot mac-clang-ninja-experimental-matlab please

File: relocate-matlab-code.sh [hunk 1]
```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+# Move a file (or glob) into the drake/mablab subdirectory.
+function matlabify {
+    dir=$(dirname $1)
+    test -d drake/matlab/$dir/
+    git mv drake/$1 drake/matlab/$dir/
+}
+
+# Twain a CMakeLists.txt into the drake/mablab subdirectory.
```
Comment (david-german-tri) [inline relocate-matlab-code.sh]: I didn't believe "twain" was a verb.  I was wrong.

File: relocate-matlab-code.sh [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+function matlabify_cmake {
+    dir=$1
+    test -f drake/$dir/CMakeLists.txt
+    cp -a drake/$1/CMakeLists.txt drake/matlab/$dir/CMakeLists.txt
+    perl -pi -e 'BEGIN { undef $/; } s|\n*# ==== Below this line is MATLAB-specific code ====.*# ==== Below this line is C.. and MATLAB shared code ====\n*|\n|s;' drake/$1/CMakeLists.txt
```
Comment (david-german-tri) [inline relocate-matlab-code.sh]: This would not handle a C-- guard correctly.

**Comment by david-german-tri** [comment]
:lgtm: for reviewable

**Comment by jwnimmer-tri** [comment]
@RussTedrake do you want to look at or experiment with anything here before I merge (assuming Jenkins passes, etc.)?

**Comment by david-german-tri** [comment]
:lgtm:

---

Review status: 0 of 461 files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3848#-:-KUZEr3dn7fvbF-17H5A:bnfp4nl)*

**Comment by jwnimmer-tri** [comment]
Jenkins decided that github didn't exist.  Let's try some again.
@drake-jenkins-bot mac-clang-ninja-experimental-matlab please
@drake-jenkins-bot mac-clang-ninja-experimental please
@drake-jenkins-bot linux-gcc-ninja-experimental-matlab please

**Comment by jwnimmer-tri** [comment]
:bee: 
@drake-jenkins-bot mac-clang-experimental-matlab please

**Comment by jwnimmer-tri** [comment]
Sigh.  AWS is having a hissy fit or something.  We will have wait to to retest until the alarm clears.

**Comment by jwnimmer-tri** [comment]
@drake-jenkins-bot mac-clang-experimental please

**Comment by jwnimmer-tri** [comment]
Hopefully fix-forward will be good enough for any undetected breaks.

**Comment by david-german-tri** [comment]
BTW, this did pass a local smoke test for me.

**Pull request merged**